draft-ietf-acme-dns-account-01.mkd

```diff
@@ -172,6 +172,37 @@ If all the above verifications succeed, then the validation is successful. If no
 
 The client SHOULD de-provision the resource record(s) provisioned for this challenge once the challenge is complete, i.e., once the "status" field of the challenge has the value "valid" or "invalid".
 
+## Errors
+
+The server SHOULD follow the guidelines set in {{!RFC8555, Section 6.7}} for error conditions that occur during challenge validation.
+
+If the server is unable to find a TXT record for the validation domain name, it MUST include the account URL it used to construct the validation domain name in a subproblem of the problem document. The subproblem MUST have a type of `urn:ietf:params:acme:error:account_url`. It MUST include an identifier field with a type of `account_url` and value of the account URL.
+
+```json
+HTTP/1.1 403 Forbidden
+Content-Type: application/problem+json
+Link: <https://example.com/acme/directory>;rel="index"
+
+{
+    "type": "urn:ietf:params:acme:error:dns",
+    "detail": "Unable to validate challenge",
+    "subproblems": [
+        {
+            "type": "urn:ietf:params:acme:error:account_url",
+            "detail": "Unable to find TXT record for validation domain name",
+            "identifier": {
+                "type": "account_url",
```

Comment:
This might be contentious, or not possible at all.

Reply:
Arguably, we don't want this field to be machine readable anyway. This is only for human-intervention debugging purposes.

Maybe we can avoid using the `identifier` field and just require use the `detail` field.

Reply:
Something like:

If the server is unable to find a TXT record for the validation domain name, it MUST include the account URL it used to construct the validation domain name in the detail field of the problem document. The problem document MUST have a type of `urn:ietf:params:acme:error:account_url`. This problem document MAY be the top level problem document, or it may be a subproblem of another problem document.